mm2src/coins/lp_coins.rs

```diff
@@ -479,11 +494,38 @@ pub struct SearchForSwapTxSpendInput<'a> {
     pub swap_unique_data: &'a [u8],
 }
 
+#[derive(Clone, Debug, Deserialize, PartialEq, Serialize)]
+pub enum PaymentInstructions {
+    #[cfg(not(target_arch = "wasm32"))]
+    Lightning(Invoice),
+}
+
+#[derive(Display)]
+pub enum PaymentInstructionsErr {
+    LightningInvoiceErr(String),
+    InternalError(String),
+}
+
+impl From<NumConversError> for PaymentInstructionsErr {
+    fn from(e: NumConversError) -> Self { PaymentInstructionsErr::InternalError(e.to_string()) }
+}
+
+#[derive(Display)]
+pub enum ValidateInstructionsErr {
+    ValidateLightningInvoiceErr(String),
+}
+
+#[cfg(not(target_arch = "wasm32"))]
+impl From<ParseOrSemanticError> for ValidateInstructionsErr {
+    fn from(e: ParseOrSemanticError) -> Self { ValidateInstructionsErr::ValidateLightningInvoiceErr(e.to_string()) }
+}
+
 /// Swap operations (mostly based on the Hash/Time locked transactions implemented by coin wallets).
 #[async_trait]
 pub trait SwapOps {
     fn send_taker_fee(&self, fee_addr: &[u8], amount: BigDecimal, uuid: &[u8]) -> TransactionFut;
 
+    #[allow(clippy::too_many_arguments)]
```

Comment:
Same as I mentioned here https://github.com/KomodoPlatform/atomicDEX-API/pull/1468#discussion_r995558593
But I think it's better to discuss whose PR will be merged first

Reply:
Any one of me or @ozkanonur can implement this in his PR and the other can copy the implementation to the other PR like I did with copying `secret_hash_algo` parts from the other PR https://github.com/KomodoPlatform/atomicDEX-API/blob/60acd0af21dda65dbe8f43ca9a1fa6994b22571d/mm2src/mm2_main/src/lp_swap/maker_swap.rs#L231-L239 what do you think @ozkanonur? It will be a straight forward merge this way

Reply:
I believe we can do this once our PRs are merged. It's simple task and we don't need to create a conflict to resolve. Let's do this as seperated task in brand-new branch after our PRs merged.

@sergeyboyko0791 Could you create an issue for this? So we can assign it to any one of us after sometime.

Reply:
Yes I agree, will do